node/node.go

```diff
@@ -112,25 +113,27 @@ type Config struct {
 // cluster. Node handles workloads and may also run as a manager.
 type Node struct {
 	sync.RWMutex
-	config           *Config
-	remotes          *persistentRemotes
-	connBroker       *connectionbroker.Broker
-	role             string
-	roleCond         *sync.Cond
-	conn             *grpc.ClientConn
-	connCond         *sync.Cond
-	nodeID           string
-	started          chan struct{}
-	startOnce        sync.Once
-	stopped          chan struct{}
-	stopOnce         sync.Once
-	ready            chan struct{} // closed when agent has completed registration and manager(if enabled) is ready to receive control requests
-	closed           chan struct{}
-	err              error
-	agent            *agent.Agent
-	manager          *manager.Manager
-	notifyNodeChange chan *api.Node // used to send role updates from the dispatcher api on promotion/demotion
-	unlockKey        []byte
+	config             *Config
+	remotes            *persistentRemotes
+	connBroker         *connectionbroker.Broker
+	role               string
+	roleCond           *sync.Cond
+	conn               *grpc.ClientConn
+	connCond           *sync.Cond
+	nodeID             string
+	started            chan struct{}
+	startOnce          sync.Once
+	stopped            chan struct{}
+	stopOnce           sync.Once
+	ready              chan struct{} // closed when agent has completed registration and manager(if enabled) is ready to receive control requests
+	closed             chan struct{}
+	err                error
+	agent              *agent.Agent
+	manager            *manager.Manager
+	notifyNodeChange   chan *api.Node        // used to send role updates from the dispatcher api on promotion/demotion
+	notifyTLSChange    chan *api.NodeTLSInfo // used to send tls info to the agent on TLS credential when the node changes its trust root
+	notifyRootCAChange chan []byte           // used to send role updates from the dispatcher api on root CA change
```

Comment:
This doesn't seem to be initialized. I think it should be created in `New` like `notifyNodeChange`.